share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml

```diff
@@ -186,43 +181,49 @@ spack:
+  # - openfoam                                                # cgal: https://github.com/spack/spack/issues/39481
+  # - openpmd-api                                             # mgard:  mgard.tpp:63:48: error: non-constant-expression cannot be narrowed from type 'int' to 'unsigned long' in initializer list [-Wc++11-narrowing]
+  # - swig@4.0.2-fortran                                      # ?
+  # - upcxx                                                   # upcxx: /opt/intel/oneapi/mpi/2021.10.0//libfabric/bin/fi_info: error while loading shared libraries: libfabric.so.1: cannot open shared object file: No such file or directory
+  # --
+  # - bricks ~cuda                                            # bricks: /opt/intel/oneapi/compiler/2024.0/bin/sycl-post-link: error while loading shared libraries: libonnxruntime.1.12.22.721.so: cannot open shared object file: No such file or directory
+  # - pdt                                                     # pdt: pdbType.cc:193:21: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
+  # - quantum-espresso                                        # quantum-espresso@7.2 /i3fqdx5: warning: <unknown>:0:0: loop not unroll-and-jammed: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering
+  # - tau +mpi +python +syscall                               # pdt: pdbType.cc:193:21: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
 
   # GPU
-  - aml +ze
   - amrex +sycl
-  - arborx +sycl ^kokkos +sycl +openmp cxxstd=17 +examples
-  - cabana +sycl ^kokkos +sycl +openmp cxxstd=17 +examples
-  - ginkgo +sycl
-  - heffte +sycl
-  - kokkos +sycl +openmp cxxstd=17 +examples
-  - kokkos-kernels build_type=Release %oneapi ^kokkos +sycl +openmp cxxstd=17 +examples
-  - petsc +sycl
-  - slate +sycl
-  - sundials +sycl cxxstd=17 +examples-install
-  - tau +mpi +opencl +level_zero ~pdt +syscall   # tau: requires libdrm.so to be installed
+  - tau +mpi +opencl +level_zero ~pdt +syscall                # requires libdrm.so to be installed
   - upcxx +level_zero
   # --
   # - hpctoolkit +level_zero            # dyninst@12.3.0%gcc: /usr/bin/ld: libiberty/./d-demangle.c:142: undefined reference to `_intel_fast_memcpy'; can't mix intel-tbb@%oneapi with dyninst%gcc
   # - warpx compute=sycl                # warpx: spack-build-wzp6vvo/_deps/fetchedamrex-src/Src/Base/AMReX_RandomEngine.H:18:10: fatal error: 'oneapi/mkl/rng/device.hpp' file not found
+  # --
+  # - aml                                                     # aml: /opt/intel/oneapi/compiler/2024.0/bin/sycl-post-link: error while loading shared libraries: libonnxruntime.1.12.22.721.so: cannot open shared object file: No such file or directory
```

Comment:
This is a bug that will be fixed in the next release. If you install intel-oneapi-compilers, it will patch sycl-post-link to have a run_path that includes the compiler/<version>/lib directory. If you are using an externally installed compiler then you need to set LD_LIBRARY_PATH to include the directory with libonnxruntime. This will happen if you source setvars.sh on compiler/latest/env/vars.sh. I know that e4s uses the oneapi container for compilers. Do you do anything to set up the compiler other than adding it to the compilers.yaml?

Reply:
You cannot use var.sh or setvars.sh because spack clears the incoming LD_LIBRARY_PATH. This compilers.yaml will set the LD_LIBRARY_PATH:
```
- compiler:
    spec: oneapi@=2024.0.2
    paths:
      cc: /opt/intel/oneapi/compiler/latest/bin/icx
      cxx: /opt/intel/oneapi/compiler/latest/bin/icpx
      f77: /opt/intel/oneapi/compiler/latest/bin/ifx
      fc: /opt/intel/oneapi/compiler/latest/bin/ifx
    flags: {}
    operating_system: ubuntu22.04
    target: x86_64
    modules: []
    environment:
      append_path:
        LD_LIBRARY_PATH: "/opt/intel/oneapi/compiler/latest/lib"
    extra_rpaths: []
```

Reply:
Thanks, I fixed here: https://github.com/UO-OACISS/e4s/commit/0997302fc4418be111ab17ff68b4c875db00c895